### PR TITLE
Update-data-source

### PR DESCRIPTION
**Pull request by yetihan**
use latest source data 201607

### CODE REVIEW

File: gb2260/division.py [hunk 1]
```diff
@@ -119,10 +119,8 @@ def stack(self):
 def make_year_key(year):
     """A key generator for sorting years."""
     if year is None:
-        return (LATEST_YEAR, 12)
+        return (LATEST_YEAR, 7)
     year = str(year)
-    if len(year) == 4:
-        return (int(year), 12)
```
Comment (tonyseek) [inline gb2260/division.py]: What is the reason you change here for?

The structure of 2-tuple should be `(year, month)`. The default month should be `12` for solo-revised years.

File: generate.py [hunk 2]
```diff
@@ -47,9 +47,10 @@ def main():
                 source, revision, code, name = line.strip().split()
                 current_dict[int(code)] = ensure_unicode(name)
 
-    # copy lasted data ,key is None;
-    last_date = max(data.keys())
-    data[None] = data[last_date]
+    # change key of  latest data,from yyyymm to None
+    latest_date = max(data.keys())
+
```
Comment (tonyseek) [inline generate.py]: Extra spaces and empty lines need to be deleted

**Review by tonyseek — CHANGES_REQUESTED** [review]
Please clean up your `*_bak` files and squash your commits. The title and description of pull request need to be clear too. Thank you.

File: gb2260/division.py (hunk 1 above)
Comment (yetihan) [inline gb2260/division.py]: All filenames of new source data are like "yyyyMM". So,there is not need to turn 'yyyy' to (yyyy,mm). Maybe, we should change the vairable name of year

Comment (yetihan) [inline gb2260/division.py]: Hi,master.I have squash my commits. Is it ok this time ? ^_^

Comment (tonyseek) [inline gb2260/division.py]: IMO, we could not presume this except it is written down in [spec](https://github.com/cn/GB2260/blob/develop/spec.md).

Comment (yetihan) [inline gb2260/division.py]: See [spec](https://github.com/cn/GB2260/blob/develop/spec.md)
In the Data Structure part,"revision: (str) The revision year (month).",
Atfer all,i think it is more safe to reserve the code to deal year.

File: generate.py (hunk 2 above)
Comment (yetihan) [inline generate.py]: This problem has been resolved.

Comment (yetihan) [inline generate.py]: anything to do?

**Pull request closed without merging**